### PR TITLE
MINOR: increase timeouts for KafkaStreamsTest

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei @ableegoldman @ijuma

File: streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ public void testStateOneThreadDeadButRebalanceFinish() throws InterruptedExcepti
 
         TestUtils.waitForCondition(
             () -> stateListener.numChanges == 2,
-            10 * 1000,
+            30_000,
```
Comment (wicknicks) [inline streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java]: nit: make a constant for this value? helps to read and understand the time unit.

**Comment by mjsax** [comment]
Updated per review comment.

**Comment by mjsax** [comment]
Rebased to pick up test fix for connect.

**Comment by abbccdda** [comment]
Sorry what's the purpose of extending timeout here? Are there flakey tests?

**Comment by mjsax** [comment]
Yes, we saw some test failures hitting those timeouts. We hope to stabilize the Jenkins build with this.

**Comment by ijuma** [comment]
Btw, have we only seen the failures with Java 8?

**Comment by mjsax** [comment]
I am not sure. Jenkins is green. I am going to merge this now.

**Comment by ijuma** [comment]
Hmm, did you get any committer review?

**Comment by ijuma** [comment]
The reason I asked is twofold:
1. If it's only with Java 8, then it may be due to a JDK bug that was fixed in JDK 8u202.
2. If the above is the case, then it may make sense to reuse the TestUtils default timeout so that it's adjustable in one place

**Comment by mjsax** [comment]
@ijuma This PR is reverted as part of #6185 falling back to default timeout. If the problem persists, we we need to investigate further. Thanks.

**Comment by ijuma** [comment]
Thanks @mjsax.

**Pull request merged**